### ISSUE TITLE
[template/use-track-by-function] allow custom trackBy-directives

### ISSUE THREAD

**Issue by L3444** [angular-eslint/angular-eslint#1323]
**Description and reproduction of the issue**

```JSON
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": "warn"
  }
}
```

```HTML
<li *ngFor="let item of items; trackByProperty: 'id'"></li>
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `5.45.1` |
| `@typescript-eslint/parser`        | `5.45.1` |
| `ESLint`                           | `8.29.0` |
| `node`                             | `18.12.1` |
| `ng-for-track-by-property`                             | `15.0.1` |

Is: the *ngFor has the ESLint warning "ESLint: Missing trackBy function in ngFor directive" even though `trackByProperty` was provided, which has the same functionality as providing a `trackBy`.

--- 

Custom directives that provide the same functionality as trackBy, for example `trackByProperty` by dependency "ng-for-track-by-property", aren't recognized by rule "@angular-eslint/template/use-track-by-function" and reported as if a trackBy was missing.

Would it be possible to allow optional configuration of the function `isNgForTrackBy` so that additional directives, not just "ngForTrackBy", are seen as a trackBy-function?

Here's an example of how these custom trackBy-directives are implemented: https://github.com/nigrosimone/ng-for-track-by-property/blob/main/projects/ng-for-track-by-property/src/lib/ng-for-track-by-property.directive.ts

**Comment by natanbr** [comment]
I'm using the same syntax and it would be very nice to customize the rule (and avoid adding the ignore comment!
```
<!-- eslint-disable-next-line @angular-eslint/template/use-track-by-function -->
<div *ngFor="let transfer of transfers; trackById"
```
